### PR TITLE
Minable display names and nouns

### PR DESCRIPTION
**Pull request by warp-core**
**Feature:*

## Feature Details
Adds display name and noun members to Minable.
By default, the display name will be the same as the unique name of the minable and the noun will be "asteroid".

Also, no longer capitalise the first letter of each word in the name every step when displaying. The default strings will be formatted this way and, in the case of custom values, the user should use their preferred format (or, match the default format for vanilla.)

Save file: 
[MinableDisplay NameNoun~Tester.txt](https://github.com/endless-sky/endless-sky/files/10465608/MinableDisplay.NameNoun.Tester.txt)

You'll need to put the example in a plugin or something to see this functionality in action for yourself.

## UI Screenshots
Before:
image

With a custom display name:
image
~~Ignore the typo.~~ 

## Usage Examples
```
minable "large plant2"
	"display name" "Large Plant"
	noun "Object"
```

## Testing Done
Target some of the asteroids in Postverta with the save file without  this change and with this change to see no difference and then with the example to see the custom name and noun in use..

### Automated Tests Added
N/A

## Performance Impact
None

### CODE REVIEW

File: source/Minable.h [hunk 1]
```diff
@@ -50,13 +50,15 @@ class Minable : public Body {
...
+	const std::string &DisplayName() const;
+	const std::string &Noun() const;
 
 	// Place a minable object with up to the given energy level, on a random
 	// orbit and a random position along that orbit.
 	void Place(double energy, double beltRadius);
 
 	// Move the object forward one step. If it has been reduced to zero hull, it
-	// will "explode" instead of moving, creating flotsam and explosion effects.
+	// will "explode" instead of moving, creating flotsam and explosion effects.o
```
Comment (warp-core) [inline source/Minable.h:61]:
```suggestion
	// will "explode" instead of moving, creating flotsam and explosion effects.
```

**Review by Hurleveur — APPROVED** [review]
looks good to me

**Comment by RisingLeaf** [comment]
If we are going with how we handled it so far, we would have a TrueName() function and a DisplayName() function. So if you do this, you should rename Name() to TrueName(), similar to the outfit display name PR

**Review by RisingLeaf — APPROVED** [review]
looks good to me

**Pull request merged**